src/main/java/org/betonquest/betonquest/quest/event/run/RunIndependentEventFactory.java

```diff
@@ -0,0 +1,36 @@
+package org.betonquest.betonquest.quest.event.run;
+
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.quest.event.Event;
+import org.betonquest.betonquest.api.quest.event.EventFactory;
+import org.betonquest.betonquest.api.quest.event.StaticEvent;
+import org.betonquest.betonquest.api.quest.event.StaticEventFactory;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.id.EventID;
+import org.betonquest.betonquest.quest.event.CallStaticEventAdapter;
+
+import java.util.List;
+
+/**
+ * Create new {@link RunIndependentEvent} from instruction.
+ */
+public class RunIndependentEventFactory implements StaticEventFactory, EventFactory {
+
+    /**
+     * Create new {@link RunIndependentEventFactory}.
+     */
+    public RunIndependentEventFactory() {
+    }
+
+    @Override
+    public StaticEvent parseStaticEvent(final Instruction instruction) throws InstructionParseException {
+        final List<EventID> eventIDS = instruction.getList(instruction.getOptional("events"), instruction::getEvent);
+        eventIDS.addAll(instruction.getList(instruction.getOptional("event"), instruction::getEvent));
```

Comment:
I think i do not understand this peace of code.

Reply:
hope it's now more readable. Basically it's adding support for both `events:` and `event:` just like it's the case for objectives or conversations.